Add 1.17 upgrade-specific note for upstream normalization. (#19181)
hashicorp/consul: website/content/docs/upgrading/upgrade-specific.mdx

Prior to Consul 1.17.0, audit logs contained timestamps on both the original log file names as well as rotated log file names.
After Consul 1.17.0, only timestamps will be included in rotated log file names.

#### Service-defaults upstream config (Enterprise)
Prior to Consul 1.17.0, [Consul-k8s upstreams](/consul/docs/k8s/annotations-and-labels#consul-hashicorp-com-connect-service-upstreams)

> that did not explicitly specify a namespace or partition would incorrectly use service-defaults configuration (such as protocol)
> from the default partition / namespace, rather than the local partition / namespace.

Reviewer: ```suggestion
that did not explicitly specify a namespace or partition would incorrectly use service-defaults configurations, such as the incorrect protocol,
from the `default` partition and namespace, rather than the local partition and namespace.
```

> This bug is fixed starting in Consul 1.17.0, and the service-defaults from the local partition and namespace will now be fetched instead.

Reviewer: ```suggestion
This bug is fixed starting in Consul 1.17.0, and the service-defaults from the local partition and namespace are now fetched instead.
```

> If using non-default partitions and namespaces with Consul-k8s, it is advised to explicitly define these fields for all upstreams and to

Reviewer: ```suggestion
If you are using non-default partitions and namespaces with Consul-k8s, we recommend explicitly defining these fields for all upstreams. Doing so
```

> ensure proper service-defaults entries exist in each namespace / partition before upgrading so that no unexpected protocol changes occur.

Reviewer: ```suggestion
ensures that proper service-defaults entries exist in each namespace and partition prior to upgrading, which prevents unexpected protocol changes.
```
